modules/s3/s3import.py

```diff
@@ -3935,6 +3936,24 @@ def parseACL(_acl):
                             *acls[rulelist[0]])
 
     # -------------------------------------------------------------------------
+    def import_font(self, url):
```

Comment:
Yes, start with a version just for this function...abstracting to a generalised function can happen later :)
NB The import_remote_csv in this module handles .Zip, so there is a generalisation possible right away within this 1 module (although I'm sure it can be applied elsewhere too)